documentation/src/main/asciidoc/cassandraclient.adoc

```diff
@@ -0,0 +1,349 @@
+They should not be coupled to allow the API to evolve independently of the storage layer.
+This is the reason why the API is using a `FruitDto` class.
+This class will be used by Quarkus to convert JSON to java object for client request and java object to JSON for the response.
+The translation is done by quarkus-resteasy extension.
+[source,java]
+----
+public class FruitDto {
+
+  private String name;
+  private String description;
+
+  public FruitDto() {}
+
+  public FruitDto(String name, String description) {
+    this.name = name;
+    this.description = description;
+  }
+  // getters and setters omitted for brevity
+}
+----
+
+One important thing to note is that having a default constructor is required by the JSON serialization layer.
+
+== Configuring the Cassandra database
+The main two properties to configure are `contact-points`, to access the Cassandra database, and `local-datacenter`, which is required by the driver.
+
+A sample configuration should look like this:
+
+[source,properties]
+----
+# configure the Cassandra client for a replica set of two nodes
+quarkus.cassandra.contact-points={cassandra_ip}:9042
+quarkus.cassandra.load-balancing-policy.local-datacenter={dc_name}
+----
+
+In this example, we are using a single instance running on localhost:
+
+[source,properties]
+----
+# configure the Cassandra client for a single instance on localhost
+quarkus.cassandra.contact-points=127.0.0.1:9042
+quarkus.cassandra.load-balancing-policy.local-datacenter=datacenter1
+----
+
+You can configure all Java driver settings using `application.conf` or `application.json` files.
+They need to be located in the resources directory of your application.
+They will be passed automatically to the underlying driver configuration mechanism.
+Settings defined in `application.properties` with the `quarkus.cassandra` prefix will have priority over settings defined in `application.conf` or `application.json`.
+
+To see the full list of settings, please refer to the {reference-conf-url}[reference.conf].
+
+== Running a Cassandra Database
+By default, `CassandraClient` is configured to access a local Cassandra database on port 9042 (the default Cassandra port).
+If you have a local running database on this port, you should assert that:
+`quarkus.cassandra.load-balancing-policy.local-datacenter` setting is equal to dc of your local Cassandra instance.
+This value can be found with `SELECT data_center FROM system.local` query.
+
+If you want to use Docker to run a Cassandra database, you can use the following command to launch one:
+[source,shell]
+----
+docker run \
+   --name local-cassandra-instance \
+   -p 7000:7000 \
+   -p 7001:7001 \
+   -p 7199:7199 \
+   -p 9042:9042 \
+   -p 9160:9160 \
+   -p 9404:9404 \
+   -d \
+   launcher.gcr.io/google/cassandra3
+----
+Note that only the 9042 port is required. All others all optional but provide enhanced features like JMX monitoring of the C* instance.
```

Comment:
Semi-nit: we've used the actual name "Cassandra" everywhere else, probably want that here instead of "C*"